qiita_db/portal.py

```diff
@@ -32,32 +34,198 @@ def get_studies_by_portal(portal):
         return {x[0] for x in
                 conn_handler.execute_fetchall(sql, [portal_id])}
 
-    def add_study_to_portal(study, portal):
-        """Adds study to given portal
+    def _check_studies(studies):
+        conn_handler = SQLConnectionHandler()
+        # Check if any study IDs given do not exist.
+        sql = "SELECT study_id from qiita.study WHERE study_id IN %s"
+        existing = [x[0] for x in conn_handler.execute_fetchall(
+            sql, [tuple(studies)])]
+        if len(existing) != len(studies):
+            bad = map(str, (set(studies).difference(existing)))
+            raise QiitaDBError("The following studies do not exist: %s" %
+                               ", ".join(bad))
+
+    def add_studies_to_portal(portal, studies):
+        """Adds studies to given portal
 
         Parameters
         ----------
+        studies : list of int
```

Comment:
The parameters list in here should be in the same order as in the function signature.

Also missing the raises section, in case that some of the provided studies do not exist.